addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -197,6 +197,9 @@ const LinkDialog = Dialog.extend({
      */
     save: function () {
         this.linkWidget.save();
+        if (!this.linkWidget.final_data) {
```

Comment:
It looks like it is the best way to fix the issue without refactoring the `_DialogLinkWidget` save method, yet I do not understand why this method returns a `Promise.reject()` when the data from `_getData` is incomplete, and `null` when the execution is complete, as well as when the input is not valid (in the `MenuEntryDialog` extension).
Maybe it would be better to refactor it so that it has consistent outputs, so that you could test from `LinkDialog` with something like :
```
const saved = this.linkWidget.save();
if (saved) {
    this.final_data = this.linkWidget.final_data;
    return this._super(...arguments);
}
```
What do you think ?

Reply:
I also really didn't like the `Promise.reject()` as I don't really understand what it's doing here. But when I first changed it it didn't work. Seems like with `MenuEntryDialog` changed as well it works!